lib/active_merchant/billing/gateways/opp.rb

```diff
@@ -357,7 +357,7 @@ def success_from(response)
 
         success_regex = /^(000\.000\.|000\.100\.1|000\.[36])/
 
-        if success_regex =~ response['result']['code']
+        if success_regex&.match?(response['result']['code'])
```

Comment:
What does the `&` signify here?

Reply:
So in all honesty, that was one of the changes that RuboCop made automatically for me. However, [this issue](https://github.com/rubocop-hq/rubocop/issues/5026) states that the `&` there provides a `nil` check so that in the event that `success_regex` is `nil` this won't error out thanks to [the safe navigation operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/).

Reply:
I'm good with the use of `match` but think we should get rid of this `&`; if `success_regex` is nil, something has gone very wrong, but the presence of the `&` will result in the check returning the `else` clause (thereby obscuring that something has gone very wrong.) 

I wonder if Rubocop would be okay with not `&`ing a constant (i.e., changing `success_regex` to `SUCCESS_REGEX`.) Otherwise, I think we should consider telling Rubocop to be mad better. :)

Reply:
This is Rubocop being defensive on the autocorrect: https://github.com/rubocop-hq/rubocop-performance/blob/7caf6f53e782d1bc889b69fdbf8116632d4e678c/lib/rubocop/cop/performance/regexp_match.rb#L261-L270

It doesn't analyze whether that local variable is always set or not, since for most cases that would be nearly impossible to reliable do anyway with Ruby :) `TYPES_IMPLEMENTING_MATCH` contains constants, which this should've been anyway IMO, and will autocorrect without the operator.